src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,492 @@
+                        if (response.ContentLength > ushort.MaxValue) throw new ArgumentOutOfRangeException("The response it's bigger than allowed");
+                        StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.GetEncoding("UTF-8"));
+                        var retString = reader.ReadToEnd();
+                    }
+                    catch (Exception e)
+                    {
+                        Log($"Failed to send the response signature to {node}, as {e.Message}", LogLevel.Warning);
+                    }
+                }).Start();
+            }
+        }
+
+        public void ProcessRequest(StoreView snapshot, ulong id, OracleRequest req)
+        {
+            Log($"Process oracle request: {req}, txid: {req.OriginalTxid}, url: {req.Url}");
+
+            var code = ProcessUrl(req.Url, out string data);
+            foreach (var (requestId, request) in NativeContract.Oracle.GetRequestsByUrl(snapshot, req.Url))
+            {
+                var result = code == OracleResponseCode.Success ? Filter(data, request.Filter) : Array.Empty<byte>();
+                var response = new OracleResponse() { Id = requestId, Code = code, Result = result };
+                var responseTx = CreateResponseTx(snapshot, response);
+                var backupTx = CreateResponseTx(snapshot, new OracleResponse() { Code = OracleResponseCode.ConsensusUnreachable, Id = requestId, Result = Array.Empty<byte>() });
+
+                Log($"Builded response tx:{responseTx.Hash} requestTx:{request.OriginalTxid} requestId: {requestId}");
+
+                ECPoint[] oraclePublicKeys = NativeContract.RoleManagement.GetDesignatedByRole(snapshot, Role.Oracle, snapshot.Height + 1);
+                foreach (var account in Wallet.GetAccounts())
+                {
+                    var oraclePub = account.GetKey().PublicKey;
+                    if (!account.HasKey || account.Lock || !oraclePublicKeys.Contains(oraclePub)) continue;
+
+                    var txSign = responseTx.Sign(account.GetKey());
+                    var backTxSign = backupTx.Sign(account.GetKey());
+                    AddResponseTxSign(snapshot, requestId, oraclePub, txSign, responseTx, backupTx, backTxSign);
+                    SendResponseSignature(requestId, txSign, account.GetKey());
+
+                    Log($"Send oracle sign data: Oracle node: {oraclePub} RequestTx: {request.OriginalTxid} Sign: {txSign.ToHexString()}");
+                }
+            }
+        }
+
+        private OracleResponseCode ProcessUrl(string url, out string response)
+        {
+            response = null;
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var uri))
+                return OracleResponseCode.Error;
+            switch (uri.Scheme)
+            {
+                case "https":
+                    try
+                    {
+                        response = Https.Request(url);
```

Comment:
Maybe create a method in `IOracleProtocol` and `OracleHttpProtocol`, and move the `try-catch` into it. Because different protocols have different exceptions. The exceptions should be handled by the protocol implementations.

```c#
OracleResponseCode Process(Uri uri, out string response)
```

Reply:
Done